src/categorical_algebra/CSets.jl

```diff
@@ -412,13 +412,36 @@ end
 map_components(f, α::LooseACSetTransformation) =
   LooseACSetTransformation(map(f, components(α)), α.type_components, dom(α), codom(α))
 
-function is_natural(α::ACSetTransformation{S}) where {S}
+"""
+Check naturality condition for a purported ACSetTransformation, α: X→Y. 
+For each hom in the schema, e.g. h: m → n, the following square must commute:
+     αₘ
+  Xₘ --> Yₘ
+Xₕ ↓  ✓  ↓ Yₕ
+  Xₙ --> Yₙ
+     αₙ 
+
+If debug is true, return a list of violations, with elements such as: 
+  (h, index i in Xₐ, Yₕ(αₘ(i)), αₙ(Xₕ(i)))
+
+This is type-unstable, so it should not be used in performance-sensitive code.
+"""
+function is_natural(α::ACSetTransformation{S}; debug::Bool=false) where {S}
```

Comment:
I think this keyword arg should be called something like `return_failures` since `debug` is a bit ambiguous and this functionality might find uses other than debugging.

Reply:
👍🏼